src/main/java/gtPlusPlus/core/recipe/RECIPES_GREGTECH.java

```diff
@@ -1101,39 +1102,85 @@ private static void distilleryRecipes() {
                 .eut(TierEU.RECIPE_MV / 2).addTo(distillationTowerRecipes);
     }
 
+    private static void thermalBoilerRecipes() {
+        Logger.INFO("Registering Thermal Boiler Recipes.");
+
+        // Special value is used for building NEI tooltip.
+        // 1: Outputs Obsidian.
+        // 2: Outputs coal.
+        // Leave unset otherwise.
+
+        // Lava
+
+        GT_Values.RA.stdBuilder()
+                .fluidInputs(FluidUtils.getLava(1000), FluidUtils.getWater(40_000 / GT_Values.STEAM_PER_WATER))
+                .fluidOutputs(FluidUtils.getPahoehoeLava(1000), FluidUtils.getSteam(40_000))
+                .itemOutputs(
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Copper, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Tin, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Silver, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Gold, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Redstone, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.GraniteBlack, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Basalt, 1),
+                        new ItemStack(Blocks.obsidian, 1, 0))
+                .outputChances(300, 200, 100, 80, 150, 400, 400, 500).specialValue(1).duration(1 * SECONDS).eut(0)
+                .addTo(thermalBoilerRecipes);
+
+        // Pahoehoe Lava
+
+        GT_Values.RA.stdBuilder()
+                .fluidInputs(FluidUtils.getPahoehoeLava(1000), FluidUtils.getWater(40_000 / GT_Values.STEAM_PER_WATER))
+                .fluidOutputs(FluidUtils.getSteam(40_000))
+                .itemOutputs(
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Electrum, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Tantalite, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Bauxite, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Ilmenite, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Tungstate, 1),
+                        GT_OreDictUnificator.get(OrePrefixes.dustImpure, Materials.Phosphate, 1),
+                        ItemUtils.getItemStackFromFQRN("appliedenergistics2:tile.BlockSkyStone", 1),
+                        new ItemStack(Blocks.obsidian, 1, 0))
+                .outputChances(50, 300, 150, 200, 100, 100, 400, 500).specialValue(1).duration(1 * SECONDS).eut(0)
+                .addTo(thermalBoilerRecipes);
+
+        /*
```

Comment:
why is this commented out? if you dont want it, remove it please.

Reply:
maybe we run it on zeta first and see if it is balance and remove this after a test ?